tests/test_rebalance.py

```diff
@@ -48,129 +32,150 @@ def _reassign(topic, partition, replicas):
-        ('t0', '2'): ['1', '2'],
-        ('t0', '3'): ['1', '2'],
-        ('t0', '4'): ['1', '2'],
-        ('t0', '5'): ['1', '2'],
-        ('t0', '6'): ['1', '2'],
-    }
-    brokers, zk = __create_zk_for_topics(distribution)
-    o = RebalanceChange(zk, brokers)
-    # broker to partitions
-    while o.run([]):
-        pass
-
-    __verify_balanced(('1', '2'), distribution)
-
-
-def test_rebalance_with_dead_brokers():
-    distribution = {
-        ('t0', '0'): ['2', '1'],
-        ('t0', '1'): ['1', '2'],
-        ('t0', '2'): ['1', '2'],
-        ('t0', '3'): ['1', '2'],
-        ('t0', '4'): ['1', '2'],
-        ('t0', '5'): ['1', '2'],
-        ('t0', '6'): ['1', '2'],
-    }
-    _, zk = __create_zk_for_topics(distribution, broker_ids=['1', '3'])
-    o = RebalanceChange(zk, ['1', '3'])
-    while o.run([]):
-        pass
-    __verify_balanced(['1', '3'], distribution)
-
-
-def test_rebalance_with_many_topics():
-    distribution = {}
-    topic_count = 1000
-    partition_count = 8
-    for i in range(0, topic_count):
-        topic = 't{}'.format(i)
-        distribution.update({(topic, str(partition)): ['1', '2', '3'] for partition in range(0, partition_count)})
-    _, zk = __create_zk_for_topics(distribution, broker_ids=['1', '2', '3', '4', '5'])
-
-    o = RebalanceChange(zk, ['1', '2', '3', '4', '5'])
-    steps = 0
-    while o.run([]):
-        steps += 1
-    __verify_balanced(['1', '2', '3', '4', '5'], distribution, allowed_delta_leaders=5, allowed_delta_total=30)
-    # Minimum rebalance steps
-    distribution_count = 30
-    assert steps == int(1 + topic_count * partition_count * (distribution_count - 1) / distribution_count)
-
-
-def test_rebalance_invoked_on_broker_list_change():
-    zk = MagicMock()
-
-    zk.get = MagicMock(side_effect=NoNodeError)
-
-    check = RebalanceOnBrokerListChange(zk, MagicMock())
-    zk.get_broker_ids.return_value = ['1', '2', '3']
-
-    assert check.check() is not None
-    assert check.check() is None
-    zk.get_broker_ids.return_value = ['1', '2', '3']
-    assert check.check() is None
-    zk.get_broker_ids.return_value = ['1', '2', '4']
-    assert check.check() is not None
-    assert check.check() is None
-
-
-def test_combine_broker_ids():
-    assert ['1'] == combine_broker_ids(['1'], 1)
-    assert ['1', '2'] == combine_broker_ids(['1', '2'], 1)
-    assert ['1', '2'] == combine_broker_ids(['1', '2'], 1)
-    assert ['1,2', '2,1'] == combine_broker_ids(['1', '2'], 2)
-
-    assert ['1,2', '1,3', '2,1', '2,3', '3,1', '3,2'] == combine_broker_ids(['1', '2', '3'], 2)
-    assert ['1,2,3', '2,1,3', '3,1,2'] == combine_broker_ids(['1', '2', '3'], 3)
-    assert [
-               '1,2,3', '1,2,4', '1,3,4',
-               '2,1,3', '2,1,4', '2,3,4',
-               '3,1,2', '3,1,4', '3,2,4',
-               '4,1,2', '4,1,3', '4,2,3',
-           ] == combine_broker_ids(['1', '2', '3', '4'], 3)
+    assert (max_total - min_total) <= 1
+
+
+class TestRebalance(unittest.TestCase):
```

Comment:
Is there a test case that checks that only leadership changes, instead of moving things partitions around?